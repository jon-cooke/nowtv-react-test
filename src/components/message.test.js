import React from 'react';
import { mount } from 'enzyme';

import Message from './message';

it('should export a function', () => {
  expect(Message).toBeInstanceOf(Function);
});

const message = {
  id: 'cd445e6d-e514-424f-ba8f-16ec842002c6',
  userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
  message: 'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
  timestamp: '2017-02-09T04:27:38Z',
  user: {
    id: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
    firstName: 'Albert',
    lastName: 'Rose',
    email: 'arosec@bbb.org',
    avatar: 'http://dummyimage.com/100x100.jpg/5fa2dd/ffffff',
    ip: '20.79.231.223',
  },
};

it('renders correctly', () => {
  const tree = mount(<Message {...message} />);
  expect(tree).toMatchSnapshot();
});

it('does not render the email by default', () => {
  const tree = mount(<Message {...message} />);
  expect(tree.contains(message.user.email)).toBe(false);
});

it('renders the email after a mouseOver event', () => {
  const tree = mount(<Message {...message} />);
  tree.simulate('mouseover');
  expect(tree.text()).toEqual(expect.stringContaining(message.user.email));
});

it('does not render the email after a mouseOut event', () => {
  const tree = mount(<Message {...message} />);
  tree.simulate('mouseover');
  tree.simulate('mouseout');
  expect(tree.text()).toEqual(expect.not.stringContaining(message.user.email));
});
