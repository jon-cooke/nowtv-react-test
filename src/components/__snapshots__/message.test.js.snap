// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Message
  id="cd445e6d-e514-424f-ba8f-16ec842002c6"
  message="Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla."
  timestamp="2017-02-09T04:27:38Z"
  user={
    Object {
      "avatar": "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
      "email": "arosec@bbb.org",
      "firstName": "Albert",
      "id": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
      "ip": "20.79.231.223",
      "lastName": "Rose",
    }
  }
  userId="fe27b760-a915-475c-80bb-7cdf14cc6ef3"
>
  <div
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div>
      <img
        alt="Albert Rose"
        height={100}
        src="http://dummyimage.com/100x100.jpg/5fa2dd/ffffff"
        width={100}
      />
      Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.
    </div>
    <div>
      2/9/2017, 4:27:38 AM
    </div>
  </div>
</Message>
`;
